Added pre and post actions to the automation, and a toggle whether to keep the light turned on

jeells102 opened the pull request:
Added pre and post actions for the automation as well as an option to turn on or off the light once the script finishes.

jeells102 committed:
added pre and post actions and toggle for turning off/on lights

jeells102 committed:
Update parabolic_alarm_script.yaml

jeells102 committed:
Update parabolic_alarm_script.yaml

steku commented on parabolic_alarm.yaml:
```diff
@@ -190,33 +220,30 @@ action:
       steps_per_minute: !input steps_per_minute
       light_timeout: 0
       target_light: !input target_light
-  - if:
```
This statement is required to not turn the light back on if you manually turned it off

steku commented on parabolic_alarm.yaml:
```diff
@@ -190,33 +220,30 @@ action:
-          steps_per_minute: !input steps_per_minute
-          light_timeout: 0
-          target_light: !input target_light
-  - if:
```
This statement is required to not turn the light back on if you manually turned it off

steku commented on parabolic_alarm.yaml:
```diff
@@ -145,42 +161,56 @@ blueprint:
           step: 1.0
           mode: slider
-      default: 5
-  source_url: https://github.com/steku/ha_cercadian_alarm/blob/main/parabolic_alarm.yaml
```
I believe source_url is required for upgrades

steku commented on parabolic_alarm.yaml:
```diff
@@ -62,6 +72,12 @@ blueprint:
           max: 60.0
           step: 1.0
           mode: slider
+    keep_light_on:
```
This functionality is already implemented by setting the timeout to 0, but this is probably more user friendly.

steku reviewed (changes requested):
please add the if statements back in so you can turn off the light to stop the automation.

jeells102 committed:
Fixed If statements missing

Fixed if statements that were missing in the automation file, added version number of 1.1.0 as per feedback from community.

jeells102 commented:
Apologies, I had a broken version and a fixed version that I was working on and copied the wrong version over. Should now be all updated. Let me know if you need/want anything else.

jeells102 committed:
Removed "version" tag as it's not allowed

Only just realised that version is not allowed. Moved it to the description instead.

steku commented on parabolic_alarm_script.yaml:
```diff
@@ -125,12 +125,14 @@ sequence:
       - condition: and
```
I didn't test this last time but the light is not turning off

![image](https://github.com/user-attachments/assets/59765181-6212-43e4-8c8a-113393b4c374)

steku replied:
This variable is not being sent to the script by the call from the automation/blueprint

jeells102 replied:
Give that a go now? I think I've fixed it but not 100% sure.

jeells102 committed:
Update parabolic_alarm_script.yaml

For toggle light on input boolean.

jeells102 committed:
Fixed keep_light_on variable in automation

jeells102 committed:
Fixed keep_light_on variable

steku commented on parabolic_alarm.yaml:
```diff
@@ -218,5 +239,9 @@ action:
           steps_per_minute: !input steps_per_minute
           light_timeout: !input light_timeout
           target_light: !input target_light
+          keep_light_on: !input keep_light_on # Add this line to pass the variable
```
you're sending it but the script has no way to recieve it. maybe remove this functionality and just put the pre and post options in until you fugure it out?

steku commented:
Please thoroughly test all code changes and execution flows before asking for a review.

example tests:
- with people
- without people
- with pre and post
- with pre but no post
- with post but no pre
- with keep light on and without
- with people and no pre or post
- without people and on pre or post
- etc

steku commented:
Added pre-post option to another PR
